posthog/warehouse/models/managed_viewset.py

```diff
@@ -0,0 +1,210 @@
+from django.db import models, transaction
+
+import structlog
+
+from posthog.hogql.database.models import DecimalDatabaseField
+
+from posthog.models.team import Team
+from posthog.models.utils import CreatedMetaFields, UpdatedMetaFields, UUIDTModel
+
+logger = structlog.get_logger(__name__)
+
+
+class ManagedViewSet(CreatedMetaFields, UpdatedMetaFields, UUIDTModel):
+    class Kind(models.TextChoices):
+        REVENUE_ANALYTICS = "revenue_analytics", "Revenue Analytics"
+
+    team = models.ForeignKey(Team, on_delete=models.CASCADE)
+    kind = models.CharField(max_length=64, choices=Kind.choices)
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=["team", "kind"],
+                name="unique_managed_viewset_team_kind",
+            )
+        ]
+
+    def sync_views(self) -> None:
```

Comment:
Some model tests would go a long way here please - e.g. checking things get created correctly and are materialized etc